master/internal/scheduler/kubernetes_resource_provider.go

```diff
@@ -251,7 +258,25 @@ func (k *kubernetesResourceProvider) receivePodStarted(ctx *actor.Context, msg s
 	handler := container.task.handler
 	handler.System().Tell(handler, ContainerStarted{Container: container})
 
-	// TODO (DET-3422): add in proxying initialization.
+	if len(msg.Ports) == 0 {
```

Comment:
non-blocking: I think it would make sense to drop this check and just go ahead and store an empty slice in the map. That simplifies the code and keeps things more consistent (in terms of what keys are present in the maps).

Reply:
done